orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -104,6 +111,11 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(
       // as the session is not initialized yet. So using macro for now.
       transformers.emplace_back(std::make_unique<BiasGeluFusion>(compatible_eps));
       transformers.emplace_back(std::make_unique<IsInfReduceSumFusion>(compatible_eps));
+
+      if (device.Type() == OrtDevice::GPU) {
+        // Execute this graph transformer only for GPU (CUDA and ROCM) device
+        ORT_THROW_IF_ERROR(rule_transformer->Register(std::make_unique<GatherReplacement>()));
```

Comment:
you can use "compatible_eps" to make this transform for CUDA only. 

e.g. cuda_rocm_eps

Reply:
These graph transformations happen before the `InferenceSession` is created. And so we don't have access to the execution providers yet. And hence, I pass in the device argument to this function.